ops/model.py

```diff
@@ -1785,6 +1786,19 @@ def __iter__(self):
     def __getitem__(self, key: Union['Unit', 'Application']) -> 'RelationDataContent':
         return self._data[key]
 
+    def __setitem__(self, key: Union['Unit', 'Application'], value: Dict[str, str]) -> None:
+        # v0 -- make a new RelationDataContent on assignment
+        content = RelationDataContent(relation=self.relation, entity=key, backend=self._backend)
+        content.update(value)
+        self._data[key] = content
```

Comment:
I feel like this will end up being problematic because people might keep references to the `RelationDataContent` object, and you could then end up with two with different values, both meant to be accurately referring to what is/will be in Juju. There isn't a simple way to say "your old object is now obsolete".